system/HTTP/Cookie/CookieInterface.php

```diff
@@ -0,0 +1,208 @@
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\HTTP\Cookie;
+
+/**
+ * Interface for a value object representation of an HTTP cookie.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+ */
+interface CookieInterface
+{
+	/**
+	 * Cookies will be sent in all contexts, i.e in responses to both
+	 * first-party and cross-origin requests. If `SameSite=None` is set,
+	 * the cookie `Secure` attribute must also be set (or the cookie will be blocked).
+	 */
+	public const SAMESITE_NONE = 'none';
+
+	/**
+	 * Cookies are not sent on normal cross-site subrequests (for example to
+	 * load images or frames into a third party site), but are sent when a
+	 * user is navigating to the origin site (i.e. when following a link).
+	 *
+	 * @var string
+	 */
+	public const SAMESITE_LAX = 'lax';
+
+	/**
+	 * Cookies will only be sent in a first-party context and not be sent
+	 * along with requests initiated by third party websites.
+	 *
+	 * @var string
+	 */
+	public const SAMESITE_STRICT = 'strict';
+
+	/**
+	 * RFC 6265 allowed values for the "SameSite" attribute.
+	 *
+	 * @var string[]
+	 *
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite
+	 */
+	public const ALLOWED_SAMESITE_VALUES = [
+		self::SAMESITE_NONE,
+		self::SAMESITE_LAX,
+		self::SAMESITE_STRICT,
+	];
+
+	/**
+	 * Expires date format.
+	 *
+	 * @var string
+	 *
+	 * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Date
+	 * @see https://tools.ietf.org/html/rfc7231#section-7.1.1.2
+	 */
+	public const EXPIRES_FORMAT = 'D, d-M-Y H:i:s T';
+
+	/**
+	 * Returns a unique identifier for the cookie consisting
+	 * of its name, prefix, domain, and path.
+	 *
+	 * @return string
+	 */
+	public function getId(): string;
+
+	/**
+	 * Checks if the cookie should be sent with no URL encoding.
+	 *
+	 * @return boolean
+	 */
+	public function isRaw(): bool;
+
+	/**
+	 * Gets the cookie prefix.
+	 *
+	 * @return string
+	 */
+	public function getPrefix(): string;
+
+	/**
+	 * Gets the cookie name.
+	 *
+	 * @return string
+	 */
+	public function getName(): string;
+
+	/**
+	 * Gets the cookie name prepended with the prefix, if any.
+	 *
+	 * @return string
+	 */
+	public function getPrefixedName(): string;
+
+	/**
+	 * Gets the cookie value.
+	 *
+	 * @return string
+	 */
+	public function getValue(): string;
+
+	/**
+	 * Gets the time in Unix timestamp the cookie expires.
+	 *
+	 * @return integer
+	 */
+	public function getExpiresTimestamp(): int;
+
+	/**
+	 * Gets the formatted expires time.
+	 *
+	 * @return string
+	 */
+	public function getExpiresString(): string;
+
+	/**
+	 * Checks if the cookie is expired.
+	 *
+	 * @return boolean
+	 */
+	public function isExpired(): bool;
+
+	/**
+	 * Gets the "Max-Age" cookie attribute.
+	 *
+	 * @return integer
+	 */
+	public function getMaxAge(): int;
+
```

Comment:
setcookie() don't use it

Reply:
I'm well aware of its signature.

Reply:
@paulbalandan but you pass it to `setcookie()` - it's unusable

Reply:
I can't find where you found I used it in `setcookie`. `CookieStore::dispatch()` uses the PHP 7.3+ signature of `setcookie` and I'm passing into the 3rd param the options array produced by `Cookie::getOptions()`.